src/components/PoolingWidget/index.tsx

```diff
@@ -128,9 +132,46 @@ function addRemoveMapItem(map: Map<number, TokenDetails>, newToken: TokenDetails
   return copyMap
 }
 
+// TODO: Decide the best place to put this. This file is too long already, but feels to specific for utils
```

Comment:
I like the comments below, also agree that the "combinations" should be done in a util